src/main/java/xbot/common/subsystems/drive/PurePursuitCommand.java

```diff
@@ -337,10 +343,24 @@ private void advancePointIfNotAtLastPoint() {
      * @param translation The desired translation power, which will be constrained by the budget.
      * @return 
      */
-    private double constrainToMotionBudget(double rotation, double translation) {
-        double remainingBudget = motionBudget.get() - Math.abs(rotation);
-        double constrainedRemainingBudget = MathUtils.constrainDouble(remainingBudget, 0, 1);
-        return translation * constrainedRemainingBudget;
+    private double constrainTranslation(double rotation, double translation, double dotProduct) {
+        if (useDotProductDriving) {
+            return translation *= dotProduct;
```

Comment:
I think it might be worth a comment here about what this does and why it works (range of values of dotProduct and under what circumstances it would be what, 0, 90, 180 etc...)